src/server/game/Spells/Spell.cpp

```diff
@@ -4973,6 +4973,23 @@ SpellCastResult Spell::CheckCast(bool strict)
         if (castResult != SPELL_CAST_OK)
             return castResult;
 
+        // If it's not a melee spell, check if vision is obscured by SPELL_AURA_INTERFERE_TARGETTING
+        if (m_spellInfo->DmgClass != SPELL_DAMAGE_CLASS_MELEE)
```

Comment:
Camouflage has it in description:

> You and your pet blend into your surroundings, becoming untargetable by ranged attacks. 

Smoke bomb does not. That is why I am not sure :)